### ISSUE TITLE
Rule object-literal-key-quotes mistakenly flags the empty string as not needing quotes

### ISSUE THREAD

**Issue by gustavderdrache** [palantir/tslint#1761]
### Bug Report

- __TSLint version__: `3.15.1`
- __TypeScript version__: `2.1.1`
- __Running TSLint via__: both grunt-tslint and VS Code

#### TypeScript code being linted

```ts
const bug = {
  '': 'bug',
};
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "object-literal-key-quotes": [true, "as-needed"]
  }
}
```

#### Actual behavior

TSLint complains:
```
const bug = {
  '': 'bug',
  ~~         [Unnecessarily quoted property '' found.]
};
```

#### Expected behavior

No error; the empty string as a property should always be quoted. I found this bug when filling out a Backbone router, the key '' corresponds to the home page.

**Comment by nchen63** [comment]
closed by #1762
